File: 58. Length of Last Word/lengthLastWord.js
var lengthOfLastWord = function (s) {
	const arr = s.trim().split(/\s+/); // trim is needed to remove leading or trailing spaces
	console.log(arr);
	return arr[arr.length - 1].length;
};

const str = '   fly me   to   the moon  ';
console.log(lengthOfLastWord(str));

/*
"Hello World" = 5
"   fly me   to   the moon  " = 4
"luffy is still joyboy" = 6
*/

File: 58. Length of Last Word/lengthLastWord.cpp
#include <iostream>
#include <string>

using namespace std;

int lengthOfLastWord(string s) {

  for (int i = s.size() - 1; i > 0; i--) {
    if (s[i] == ' ') {
      s.pop_back();
    } else
      break;
  }

  for (int i = s.size(); i > 0; i--) {
    if (s[i - 1] == ' ')
      return s.size() - i;
  }
  return s.size();
}

int main() {
  string str = "";
  cout << lengthOfLastWord(str);
}

/*
"Hello World" = 5
"   fly me   to   the moon  " = 4
"luffy is still joyboy" = 6
*/